exec/csnex/Support.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace csnex
+{
+    public class Support
+    {
+        static public List<string> Paths;
+
+        public Support(string source_path)
+        {
+            Paths = new List<string>();
+            // Always check current directory;
+            Paths.Add(string.Format(".{0}", Path.DirectorySeparatorChar));
+            // Also check the Neon executable path.
+            Paths.Add(string.Format("{0}{1}", source_path, Path.DirectorySeparatorChar));
+            // Then check the cwd.
+            Paths.Add(Environment.CurrentDirectory);
+
+            // Get paths from %NEONPATH% environment variable, if there is one.
+            string ne = Environment.GetEnvironmentVariable("NEONPATH");
+            if (ne != null) {
+                string[] evpaths = ne.Split(Path.PathSeparator);
+                foreach (string s in evpaths) {
+                    // Don't bother pushing empty paths.
+                    if (s != null && s.Length > 0) {
+                        Paths.Add(s);
+                    }
+                }
+            }
+            // Next, check for .neonpath, and process any paths that might exist in it.
+            if (File.Exists(".neonpath")) {
```

Comment:
This sort of existence check is not robust code, it is a potential race condition. It's possible that the file could be removed by something else between the time you check the existence and try to open it. Instead, simply try to open the file and handle any failure to open in the usual way.

Reply:
Use a try/catch for this now.